test/OpenTelemetry.Instrumentation.EntityFrameworkCore.Tests/EntityFrameworkDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// <copyright file="EntityFrameworkDiagnosticListenerTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Moq;
+using OpenTelemetry.Instrumentation.EntityFrameworkCore.Implementation;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.EntityFrameworkCore.Tests
+{
+    public class EntityFrameworkDiagnosticListenerTests : IDisposable
+    {
+        private readonly DbContextOptions<ItemsContext> contextOptions;
+        private readonly DbConnection connection;
+
+        public EntityFrameworkDiagnosticListenerTests()
+        {
+            this.contextOptions = new DbContextOptionsBuilder<ItemsContext>()
+                .UseSqlite(CreateInMemoryDatabase())
+                .Options;
+
+            this.connection = RelationalOptionsExtension.Extract(this.contextOptions).Connection;
+
+            this.Seed();
+        }
+
+        [Fact]
+        public void EntityFrameworkContextEventsInstrumentedTest()
+        {
+            var activityProcessor = new Mock<ActivityProcessor>();
+            using var shutdownSignal = Sdk.CreateTracerProvider(b =>
+            {
+                b.AddProcessorPipeline(c => c.AddProcessor(ap => activityProcessor.Object));
+                b.AddEntityFrameworkInstrumentation();
```

Comment:
if we enable SqlClient instrumentation and use SqlClient inside EF, do we get duplicate info - one from EF and one from SqlClient?

Reply:
Yes in that case you get an outer span from EF instrumentation and an inner span from Sql instrumentation. I just tested to be sure.

Reply:
Thanks for confirming. This is important to be noted in the readme.md for EF instrumentation and SqlClient instrumentation.

Reply:
Also unit test this, to make nothing unexpected happens. Expected is 2 spans EF outer span and Sql inner span.